packages/backend/src/graphql/resolvers/wallet_address.ts

```diff
@@ -52,11 +58,14 @@ export const getWalletAddresses: QueryResolvers<ApolloContext>['walletAddresses'
     }
   }
 
-export const getWalletAddress: QueryResolvers<ApolloContext>['walletAddress'] =
+export const getWalletAddress: QueryResolvers<TenantedApolloContext>['walletAddress'] =
   async (parent, args, ctx): Promise<ResolversTypes['WalletAddress']> => {
     const walletAddressService = await ctx.container.use('walletAddressService')
     const walletAddress = await walletAddressService.get(args.id)
-    if (!walletAddress) {
+    if (
+      !walletAddress ||
+      !tenantIdToProceed(ctx.isOperator, ctx.tenant.id, walletAddress.tenantId)
```

Comment:
We shouldn't need this `tenantIdToProceed`. We should be able to just use `forTenantId` set by the middleware. The idea being the middleware picks the correct tenant id but beyond that the service does the access control.

Reply:
You're suggesting passing the tenantId into the `await walletAddressService.get(args.id)`, correct?

(Also, currently, the `setForTenantIdGraphQLMutationMiddleware` only runs for mutations. So we would pass in  `ctx.tenant.id` instead of `forTenantId`)

Reply:
@BlairCurrey, to add to what @mkurapov described, it also ensures that:
- We are avoiding sending the tenantId to the service layer
- Making use of a single index instead of both `tenantId` and `walletId` for the lookup
- We agreed to rather return `UnknownWalletAddress` for the queries (keeping the functionality consistent to what it was before), where middleware would always throw `BadUserInput` with message: `Assignment to the specified tenant is not permitted`

Reply:
@koekiebox based on our original call, I was also thinking we _would_ pass in `tenantId` in the `walletAddressService.get(id, tenantId?)` function. Since we'll likely need it in other service methods such as incoming payment creation (where we will pass in tenantId in the `walletAddressService.get`)

Reply:
Ah, got it @mkurapov , I see you also did that with the asset. Updating ♻️ .